### PR TITLE
[Core] Validate embedding prefill tensor shapes before padding copy

### PR DESCRIPTION
**Pull request by egalli**
### Details:
EmbeddingInferRequest::infer_prefill() only checked the sequence-length dimension against NPUW_LLM_MAX_PROMPT_LEN. Since the input_ids and attention_mask ports are dynamic, no rank or total-size validation ran before the right-aligned std::copy_n padding into the static prefill tensors, so a rank-3 or multi-batch tensor could pass the length check and write out of bounds.

Assert rank 2 and batch size 1 for both tensors before the length check.

### Tickets:
- [CVS-194189](https://jira.devtools.intel.com/browse/CVS-194189)

### AI Assistance:
- AI assistance used: yes
- Bug root cause was found by AI. Fix and unit-test were generated by AI. Logic validation and manual checks were performed by developer.

### CODE REVIEW

File: src/core/reference/include/openvino/reference/mvn.hpp
Comment (praasz) [inline src/core/reference/include/openvino/reference/mvn.hpp]: This function could be removed there is util which do normalization.

File: src/core/src/op/mvn.cpp [hunk 1]
```diff
@@ -106,6 +108,23 @@ void ov::op::v6::MVN::validate_and_infer_types() {
...
+                                          "Axis ",
+                                          axis,
+                                          " is out of the tensor rank range [",
+                                          -rank_value,
+                                          ", ",
+                                          rank_value - 1,
+                                          "].");
+                }
+            }
+        }
```
Comment (praasz) [inline src/core/src/op/mvn.cpp]:
```suggestion
        if (data_rank.is_static()) {
            if (const auto axes_constant = ov::util::get_constant_from_source(input_value(1))) {
                ov::util::validate_axes(axes_constant->cast_vector<int64_t>(), data_rank, *this);
            }
        }
```


Also update the evaluate_mvn to:
```cpp
bool evaluate_mvn(ov::TensorVector& outputs,
                  const ov::TensorVector& inputs,
                  bool normalize_variance,
                  float eps,
                  ov::op::MVNEpsMode eps_mode,
                  const ov::Node& node) {
    // Computed once here, outside the per-type template, to avoid duplicating axis validation in every instantiation.
    const auto reduction_axes = ov::util::try_get_normalized_axis_set(inputs[1], inputs[0].get_shape().size(), node);
    bool rc = true;
    switch (inputs[0].get_element_type()) {
        OPENVINO_TYPE_CASE(evaluate_mvn, f32, outputs, inputs, reduction_axes, normalize_variance, eps, eps_mode);
    default:
        rc = false;
        break;
    }
    return rc;
}
```

File: src/core/tests/eval.cpp [hunk 2]
```diff
@@ -4295,4 +4296,53 @@ TEST(eval, interpolate_padding_overflow) {
 
     OV_EXPECT_THROW(std::ignore = op->evaluate(outputs, inputs), ov::Exception, _);
 }
+
+TEST(eval, mvn_6_out_of_range_positive_axis) {
```
Comment (praasz) [inline src/core/tests/eval.cpp]: Add these cases in template backend and don't update the this file.

File: src/plugins/template/backend/ops/mvn.cpp [hunk 3]
```diff
@@ -19,35 +19,15 @@ bool evaluate(const std::shared_ptr<ov::op::v0::MVN>& op, ov::TensorVector& outp
...
     auto rank = inputs[0].get_shape().size();
     if (inputs[1].get_element_type() == ov::element::i64) {
-        reduction_axes = mvn_6_axes::mvn_6_reduction_axes<int64_t>(inputs[1], rank);
+        reduction_axes = ov::reference::mvn_6_reduction_axes<int64_t>(inputs[1], rank);
     } else if (inputs[1].get_element_type() == ov::element::i32) {
-        reduction_axes = mvn_6_axes::mvn_6_reduction_axes<int32_t>(inputs[1], rank);
+        reduction_axes = ov::reference::mvn_6_reduction_axes<int32_t>(inputs[1], rank);
     } else {
         OPENVINO_THROW("Unexpected indices type");
     }
```
Comment (praasz) [inline src/plugins/template/backend/ops/mvn.cpp]: It can be simplified by:
`const auto reduction_axes = ov::util::try_get_normalized_axis_set(inputs[1], inputs[0].get_shape().size(), *op);`

**Review by praasz — CHANGES_REQUESTED** [review]
